From fc190114b68498c4a795eb2b6543893a1c53a3ef Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 28 Apr 2019 22:27:58 +0300
Subject: [PATCH] [SW-0] preboot: use API for update pending reg

Instead of playing the include game in Makefile, use API function for
receiving the device specific "update pending" register offset.

Note: we still play the include game.

Change-Id: I756095fd59e5384d21cd08076f1b0740380cd98f
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/preboot.c b/common/preboot.c
index 84de797..2cc3072 100644
--- a/common/preboot.c
+++ b/common/preboot.c
@@ -36,20 +36,19 @@
 	unsigned int next_image, next_image_stable, next_image_modifiable;
 	bool try_new_image = false;
 	char *spl_image = "candidate";
+	u64 upd_pending_reg = get_upd_pending_sts_reg();
 
-	modifiable_image_pending =
-			regmap_readl(NON_RST_FLOPS_3);
+	modifiable_image_pending = regmap_readl(upd_pending_reg);
 	if (modifiable_image_pending == FLOPS_UPD_PENDING) {
 		next_image_modifiable =
 			(*(unsigned int *)(HL_SPI_BASE + 0x60004) & 0xff) - '1';
 		if (next_image_modifiable < 2) {
 			next_image = next_image_modifiable;
 			regmap_writel((next_image + FLOPS_UPD_NXT_IMG1),
-					NON_RST_FLOPS_3);
+					upd_pending_reg);
 			try_new_image = true;
 		} else
-			regmap_writel(FLOPS_UPD_NOOP,
-					NON_RST_FLOPS_3);
+			regmap_writel(FLOPS_UPD_NOOP, upd_pending_reg);
 	}
 
 	if (!try_new_image) {
diff --git a/common/preboot.h b/common/preboot.h
index 7995125..ee3aef2 100644
--- a/common/preboot.h
+++ b/common/preboot.h
@@ -45,6 +45,7 @@
 }
 
 u64 get_cpu_boot_status_reg(void);
+u64 get_upd_pending_sts_reg(void);
 void hl_uart_init(void);
 u32 hl_spi_baudr_get(void);
 
diff --git a/gaudi/gaudi.c b/gaudi/gaudi.c
index cb71a2a..7459bab 100644
--- a/gaudi/gaudi.c
+++ b/gaudi/gaudi.c
@@ -37,6 +37,11 @@
 	return mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS;
 }
 
+u64 get_upd_pending_sts_reg(void)
+{
+	return mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_3;
+}
+
 u64 hl_pll_get_cpu_pll_base_off(void)
 {
 	return mmPSOC_CPU_PLL_BASE;
diff --git a/gaudi/gaudi.h b/gaudi/gaudi.h
index 44fc1c5..768149c 100644
--- a/gaudi/gaudi.h
+++ b/gaudi/gaudi.h
@@ -32,5 +32,3 @@
 #define PCI_DIV2_CLOCK			(830000000 / 5)
 #define UART_BAUDDIV			(PCI_DIV2_CLOCK / 115200)
 #define SPI_BAUDR			(PCI_DIV2_CLOCK / 25000000)
-
-#define NON_RST_FLOPS_3			mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_3
diff --git a/goya/goya.c b/goya/goya.c
index 293b9f9..3f00fde 100644
--- a/goya/goya.c
+++ b/goya/goya.c
@@ -163,6 +163,11 @@
 	return mmPSOC_GLOBAL_CONF_WARM_REBOOT;
 }
 
+u64 get_upd_pending_sts_reg(void)
+{
+	return mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3;
+}
+
 u64 hl_pll_get_cpu_pll_base_off(void)
 {
 	return mmCPU_PLL_BASE;
diff --git a/goya/goya.h b/goya/goya.h
index 838aaa4..afdf84c 100644
--- a/goya/goya.h
+++ b/goya/goya.h
@@ -10,5 +10,3 @@
 
 #define mmCPU_PLL_BASE		0x4A2000
 #define mmIC_PLL_BASE		0x4A3000
-
-#define NON_RST_FLOPS_3		mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3
